src/CacheInterface.php

```diff
@@ -34,9 +34,10 @@ public function get($key, $default = null);
      *
      * @param string $key
      * @param mixed $value
+     * @param int|null $ttl
```

Comment:
Should this use `float` instead of `int` seconds?

Reply:
If storing things in a cache for sub second times then yes, not sure how often that happens though

Reply:
Admittedly, my use cases would be fulfilled by having just second precision.

However, at the very least this PR would have to be updated to use `microtime(true)` instead of `time()` everywhere, so that storing an item at 09:49.999 with `2s` TTL would expire at 09:51.999 instead of `09:51.000` already.

At this point I would argue we might as well accept a float TTL (also for consistency with react/event-loop). But given I don't really have a use case that requires this right now, I guess I'm okay with this either way :man_shrugging:

Reply:
Ok changed it to `float` and `microtime(true)`